### ISSUE TITLE
Access request data from inside JSON schema or keyword validator

### ISSUE THREAD

**Issue by purplepenguin2** [fastify/help#1007]
## 💬  Question here 

I'd like to use data from an earlier hook as part of the schema for a later route.  Is this possible to do?  For example:

```js
fastify.post(
  '/',
  {
    schema: {
      body: {
        type: 'object',
        properties: {
          myCoolProp: { enum: req.arrayValueFromEarlierHook },
        },
        required: ['myCoolProp'],
      },
    },
  },
  async (req, res) => {
    // bleh
  },
);
````

I know I could also do the validation manually in the route handler itself, but it'd be nice to get the built-in error messaging, etc. by using a schema.  I realize it's not possible to simply declare it the way I did in the example above since they're compiled on server start, but the example is just to get the use case across.  I was looking at adding a custom keyword as well, but it doesn't appear the keyword validate function has access to the request object either.  Is there a straightforward way to achieve this, or is this simply not possible?

## Your Environment

- *node version*: latest
- *fastify version*: latest
- *os*: Mac and Linux

**Comment by mcollina** [comment]
It's possible but it'd be complex and non performant, as well as probably unsafe.

**Comment by purplepenguin2** [comment]
Yeah, I figured it would take a performance hit, and the input is coming from a validated source, so it's not random from a user, but fair enough. I didn't think it would be simple, but wanted to triple check I wasn't missing anything obvious.  To close this out, I'm assuming your recommendation would simply be to do the check in the route itself and not as part of the schema? Thanks.
